common/src/stack/command/stack/commands/sync/dns/__init__.py

```diff
@@ -28,4 +25,5 @@ def run(self, params, args):
 
 		self.notify('Sync DNS')
 		self.runPlugins()
-		subprocess.call(['systemctl','restart','named'])
+		if self.call('list.network', ['dns=true']):
+			self._exec('systemctl restart named'.split())
```

Comment:
I'm not sure whether to commend you or scold you for being clever and not just writing a list of strings.